src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K,V>[] table;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int bucket = (table.length - 1) & hash(newNode.key);
+        Node<K, V> node = table[bucket];
+        if (node == null) {
+            table[bucket] = newNode;
+        } else {
+            while (node.next != null || Objects.equals(node.key, newNode.key)) {
+                if (Objects.equals(node.key, newNode.key)) {
+                    node.value = newNode.value;
+                    return;
+                }
+                node = node.next;
+            }
+            node.next = newNode;
+        }
+        if (++size > threshold) {
```

Comment:
why do we use pre incrementation instead of post ?

Reply:
done